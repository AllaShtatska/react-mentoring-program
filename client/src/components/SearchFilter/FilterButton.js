import styled from "styled-components";

const FilterButton = styled.button`
  background: ${props => (props.selected ? "brown" : "grey")};
`;

export default FilterButton;
