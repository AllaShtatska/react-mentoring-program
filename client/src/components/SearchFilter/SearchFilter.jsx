import '../../static/inline.css';
import '../../static/filter-button.css';

import React from 'react';
import SEARCH_FILTER_TYPE from './searchFilterType'

class SearchFilter extends React.Component {
    get className() {
        const isFilterSelected = this.props.currentFilter === this.props.filter;
        
        return isFilterSelected ? 'filter-button-selected' : 'filter-button-deselected';
    }

    getFilterTitle() {
        switch (this.props.filter) {
            case SEARCH_FILTER_TYPE.genre: return "Genre";
            case SEARCH_FILTER_TYPE.title:return "Title";
        }
    }

    render() {
        return (
            <div className="inline">
                <button className={this.className} onClick={() => this.props.onSwitchFilter(this.props.filter)}>{this.getFilterTitle()}</button>
            </div>
        );
    }
}

export default SearchFilter;