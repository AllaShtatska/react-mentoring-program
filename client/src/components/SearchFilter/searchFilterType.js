const searchFilterType={
    title: "title",
    genre: "genre"
};

export default searchFilterType;