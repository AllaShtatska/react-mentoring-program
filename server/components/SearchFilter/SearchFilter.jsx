import inlineStyles from "../../static/inline.css";
import filterButtonStyles from "../../static/filter-button.css";

import React from "react";
import SEARCH_FILTER_TYPE from "./searchFilterType";

class SearchFilter extends React.Component {
  get className() {
    const isFilterSelected = this.props.currentFilter === this.props.filter;

    return isFilterSelected
      ? filterButtonStyles["filter-button-selected"]
      : filterButtonStyles["filter-button-deselected"];
  }

  getFilterTitle() {
    switch (this.props.filter) {
      case SEARCH_FILTER_TYPE.genre:
        return "Genre";
      case SEARCH_FILTER_TYPE.title:
        return "Title";
    }
  }

  render() {
    return (
      <div className={inlineStyles.inline}>
        <button
          className={this.className}
          onClick={() => this.props.onSwitchFilter(this.props.filter)}
        >
          {this.getFilterTitle()}
        </button>
      </div>
    );
  }
}

export default SearchFilter;
