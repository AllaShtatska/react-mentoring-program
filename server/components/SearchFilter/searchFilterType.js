import keyMirror from 'keymirror';

const SEARCH_FILTER_TYPE = keyMirror({
    title: null,
    genre: null,
  });

export default SEARCH_FILTER_TYPE;