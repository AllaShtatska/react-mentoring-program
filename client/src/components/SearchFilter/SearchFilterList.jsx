import React from 'react';
import PropTypes from 'prop-types';
import SEARCH_FILTER_TYPE from './searchFilterType'
import SearchFilter from './SearchFilter'


const SearchFilterList = ({ filter, onSwitchFilter }) => {
    const searchFilters = Object.values(SEARCH_FILTER_TYPE).map(searchFilter => (
      <SearchFilter
        key={searchFilter}
        currentFilter={filter}
        filter={searchFilter}
        onSwitchFilter={onSwitchFilter}
     />
    ));
    
    return(
      <div className="inline">
         Search by
        { searchFilters }
      </div>
    );
  };

  SearchFilterList.propTypes = {
    filter: PropTypes.string.isRequired,
    onSwitchFilter: PropTypes.func.isRequired,
  };

export default SearchFilterList;