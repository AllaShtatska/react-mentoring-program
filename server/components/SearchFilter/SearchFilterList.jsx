import React from "react";
import PropTypes from "prop-types";

import SEARCH_FILTER_TYPE from "./searchFilterType";
import SearchFilter from "./SearchFilter";
import inlineStyles from "../../static/inline.css";

const SearchFilterList = ({ filter, onSwitchFilter }) => {
  const searchFilters = Object.values(SEARCH_FILTER_TYPE).map(searchFilter => (
    <SearchFilter
      key={searchFilter}
      currentFilter={filter}
      filter={searchFilter}
      onSwitchFilter={onSwitchFilter}
    />
  ));

  return (
    <div className={inlineStyles.inline}>
      Search by
      {searchFilters}
    </div>
  );
};

SearchFilterList.propTypes = {
  filter: PropTypes.string.isRequired,
  onSwitchFilter: PropTypes.func.isRequired
};

export default SearchFilterList;
