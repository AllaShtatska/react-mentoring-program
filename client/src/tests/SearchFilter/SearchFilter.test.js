import React from 'react';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';
import SearchFilter from '../../components/SearchFilter/SearchFilter';
import SEARCH_FILTER_TYPE from '../../components/SearchFilter/searchFilterType';

it ('renders correctly if title is selected as a filter', ()=>{
    const component = renderer.create(<SearchFilter currentFilter={SEARCH_FILTER_TYPE.title} filter={SEARCH_FILTER_TYPE.title} onSwitchFilter={function(){}}/>);

    var jsonComponent = component.toJSON();
    expect(jsonComponent).toMatchSnapshot();
});

it ('should trigger switch filter on click', ()=>{
    const switchFilter = jest.fn();
    const component = shallow(<SearchFilter currentFilter={SEARCH_FILTER_TYPE.title} filter={SEARCH_FILTER_TYPE.genre} onSwitchFilter={switchFilter}/>);

    component.find('button').simulate('click');

    expect(switchFilter).toHaveBeenCalled();
    expect(switchFilter).toBeCalledWith(SEARCH_FILTER_TYPE.genre);
});