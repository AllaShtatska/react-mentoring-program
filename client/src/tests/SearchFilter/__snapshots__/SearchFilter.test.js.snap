// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly if option title is selected 1`] = `
<div
  className="inline"
>
  <button
    className="filter-button-selected"
    onClick={[Function]}
  >
    Title
  </button>
</div>
`;

exports[`renders correctly if title is selected as a filter 1`] = `
<div
  className="inline"
>
  <button
    className="filter-button-selected"
    onClick={[Function]}
  >
    Title
  </button>
</div>
`;
